src/ping.c

```diff
@@ -565,8 +574,38 @@ static int ping_config(const char *key, const char *value) /* {{{ */
   return 0;
 } /* }}} int ping_config */
 
-static void submit(const char *host, const char *type, /* {{{ */
-                   gauge_t value) {
+static void submit_distribution(const char *host, const char *type,
+                                distribution_t *dist) {
+  metric_family_t *fam = calloc(1, sizeof(*fam));
+  if (fam == NULL) {
+    free(fam);
```

Comment:
Why are you freeing metric_family_t in case then fam is NULL? I think this line is redundant